### PR TITLE
Sort a given sequence numbers

### PR DESCRIPTION
**Pull request by giangle-unikie**
The program will sort given numbers, if there is any value that is not a number in the list, the program will print an error message and exit.

### CODE REVIEW

File: python/sort_number.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import sys
+#This function will sort a given sequence of numbers
+#if value in the list of numbers is not interger or float, program will exit
+def sort_numbers(numbers):
+    for value in numbers:
+        check_value_int = isinstance(value, int)
```
Comment (milva-unikie) [inline python/sort_number.py]: 'isinstance()' can be given multiple types, so I don't think there is need to check them separately in this case.

`isinstance(value, (int, float))` should work just the same.

File: python/sort_number.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        check_value_float = isinstance(value, float)
+
+        if not check_value_int and not check_value_float:
+            print("Error: " + str(value) + " is not integer or float.")
```
Comment (milva-unikie) [inline python/sort_number.py]: I would prefer using comma in prints. With comma there is no need to do type changes or add additional spaces before and after the variable
```suggestion
            print("Error:", value, "is not integer or float.")
```

File: python/sort_number.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            sys.exit(1)
+
+    numbers.sort()
+    return numbers
+
+def main():
+    numbers = [3.9, 1.16, 4, 111, 5, 9, 2, 6, 5, 3, 5.0001]
```
Comment (milva-unikie) [inline python/sort_number.py]: Testing the program would be easier it there would be a option to input the list either as an argument to the program or as an input asked in main (even though the assignment does not specify this clearly).

File: python/sort_number.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import sys
+#This function will sort a given sequence of numbers
```
Comment (milva-unikie) [inline python/sort_number.py:4]: It is recommended in the Python guidelines that between the imports and every function, there is two empty lines.

**Review by milva-unikie — CHANGES_REQUESTED** [review]
A few comments. I did not actually test my code suggestion so apologies if there are some syntax errors in them.

File: python/sort_number.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import sys
+
+
+#This function will sort a given sequence of numbers
+#if value in the list of numbers is not interger or float, program will exit
+def sort_numbers(numbers):
+    for value in numbers:
+        check_value_number = value.replace('.', '', 1).isdigit()
```
Comment (milva-unikie) [inline python/sort_number.py]: 'check_value_number' is pretty unclear variable name, I think for example 'value_is_number' would make more sense.

File: python/sort_number.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        if not check_value_number:
+            print("Error: " ,value, " is not integer or float.")
```
Comment (milva-unikie) [inline python/sort_number.py]: As I mentioned in my last comments, no need for the spaces inside the quotes before and after the variable if using commas. Python adds the spaces automatically.
```suggestion
            print("Error:", value, "is not integer or float.")
```

**Review by milva-unikie — CHANGES_REQUESTED** [review]
This new version can not handle negative numbers

> Enter the numbers separated by space: -2
> Error:  -2  is not integer or float.

File: python/sort_number.py (hunk 5 above)
Comment (giangle-unikie) [reply python/sort_number.py]: Ok, I will change it.

**Comment by jounraun** [comment]
$ python3 sort_number.py 
Enter the numbers separated by space: 8 6 22 888 -1 2222
['-1', '22', '2222', '6', '8', '888']

**Comment by jourau** [comment]
ok

**Pull request closed without merging**